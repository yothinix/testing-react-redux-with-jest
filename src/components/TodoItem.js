import React, { PropTypes } from 'react'
import style from './TodoItem.css'

const TodoItem = ({ text, complete }) => (
  <li className={complete ? style.complete : ''}>{text}</li>
)

TodoItem.propTypes = {
  text: PropTypes.string.isRequired,
  complete: PropTypes.bool,
}

TodoItem.defaultProps = {
  complete: false,
}

export default TodoItem

