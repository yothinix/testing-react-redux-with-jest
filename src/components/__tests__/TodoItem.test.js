import React from 'react'
import { shallow } from 'enzyme'
import TodoItem from '../TodoItem'

describe('TodoItem', () => {
  it('renders correct structure', () => {
    const component = shallow(<TodoItem text="Hello" />)

    expect(component.is('li')).toBe(true)
  })
})
