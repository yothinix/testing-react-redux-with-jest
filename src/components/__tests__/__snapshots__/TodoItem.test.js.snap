// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem matches its snapshot - complete 1`] = `
<li
  className="complete"
>
  complete
</li>
`;

exports[`TodoItem matches its snapshot - not complete 1`] = `
<li
  className=""
>
  Not complete
</li>
`;
